share/github-backup-utils/ghe-backup-es-audit-log

```diff
@@ -26,31 +26,34 @@ else
   es_port=9200
 fi
 
-if ! indices=$(ghe-ssh "$host" "curl -s \"localhost:$es_port/_cat/indices/audit_log*?h=index"\"); then
+if ! indices=$(ghe-ssh "$host" "curl -s \"localhost:$es_port/_cat/indices/audit_log*?h=index,pri.store.size&bytes=b\""); then
   echo "Error: failed to retrieve audit log indices." 1>&2
   exit 1
 fi
 
+# Exit if no indices were found
+[ -z "$indices" ] && exit
+
 # Determine if the audit log migration has occurred or is needed.
 if echo 'set -o pipefail; ! test -e /data/user/common/es-scan-complete && test -f /usr/local/share/enterprise/run-audit-log-transitions.sh' | ghe-ssh "$host" /bin/bash; then
   if echo 'set -o pipefail; echo n | /usr/local/share/enterprise/run-audit-log-transitions.sh > /dev/null 2>&1 && touch /data/user/common/es-scan-complete' | ghe-ssh "$host" /bin/bash; then
     touch $GHE_SNAPSHOT_DIR/es-scan-complete
   fi
 fi
 
-current_index="audit_log(-[0-9]+)?-$(ghe-ssh "$host" 'date +"%Y-%m"')(-[0-9]+)?"
+while read index; do
+  set $index
+  index_name=$1
+  index_size=$2
 
-for index in $indices; do
-  if [[ -f $GHE_DATA_DIR/current/audit-log/$index.gz && -f $GHE_DATA_DIR/current/audit-log/$index.gz.complete && ! $index =~ $current_index ]]; then
-    # Hard link any older indices that are complete, since these won't change
-    ln $GHE_DATA_DIR/current/audit-log/$index.gz $GHE_SNAPSHOT_DIR/audit-log/$index.gz
-    ln $GHE_DATA_DIR/current/audit-log/$index.gz.complete $GHE_SNAPSHOT_DIR/audit-log/$index.gz.complete
```

Comment:
As we're switching methods, I think it's worth cleaning up all the old and now unused `*.complete` files here whilst we're at it. They take no space, but will unnecessarily occupy inodes once this PR ships.

Reply:
I didn't worry about this for a couple of reasons:

These will cycle out eventually, as only older snapshots will contain these.
Modifying previous backup snapshots is something we have generally avoided (apart from pruning them completely), so I didn't want to set a new precedent.